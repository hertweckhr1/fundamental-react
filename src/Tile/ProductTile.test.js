import { mount } from 'enzyme';
import ProductTile from './ProductTile';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<ProductTile />', () => {
    const productMediaTile = (
        <ProductTile className='pink' isButton>
            <ProductTile.Media image='https://techne.yaas.io/images/product-thumbnail-wide.png' />
            <ProductTile.Content title='Tile Title'>
                <p>Tile Description</p>
            </ProductTile.Content>
        </ProductTile>
    );

    const disabledProductMediaTile = (
        <ProductTile disabled>
            <ProductTile.Media
                className='blue'
                image='https://techne.yaas.io/images/product-thumbnail-wide.png' />
            <ProductTile.Content className='blue' title='Tile Title'>
                <p>Tile Description</p>
            </ProductTile.Content>
        </ProductTile>
    );

    test('create tile component', () => {
        // product media tile
        let component = renderer.create(productMediaTile);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // disabled product media tile
        component = renderer.create(disabledProductMediaTile);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the ProductTile component', () => {
            const element = mount(<ProductTile data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
