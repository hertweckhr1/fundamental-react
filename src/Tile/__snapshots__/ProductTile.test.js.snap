// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTile /> create tile component 1`] = `
<div
  className="fd-product-tile pink"
  isButton={true}
>
  <div
    className="fd-product-tile__media"
    style={
      Object {
        "backgroundImage": "url(https://techne.yaas.io/images/product-thumbnail-wide.png)",
      }
    }
  />
  <div
    className="fd-product-tile__content"
  >
    <h2
      className="fd-product-tile__title"
    >
      Tile Title
    </h2>
    <p>
      Tile Description
    </p>
  </div>
</div>
`;

exports[`<ProductTile /> create tile component 2`] = `
<div
  className="fd-product-tile is-disabled"
>
  <div
    className="fd-product-tile__media blue"
    style={
      Object {
        "backgroundImage": "url(https://techne.yaas.io/images/product-thumbnail-wide.png)",
      }
    }
  />
  <div
    className="fd-product-tile__content blue"
  >
    <h2
      className="fd-product-tile__title"
    >
      Tile Title
    </h2>
    <p>
      Tile Description
    </p>
  </div>
</div>
`;
